SparkPost/2web2ui
# FE-716 - Add distinct integration tests, set up Travis CI to run said…

[FE-716](https://jira.int.messagesystems.com/browse/FE-716)

Also addresses:

- [FE-780](https://jira.int.messagesystems.com/browse/FE-780)
- [FE-781](https://jira.int.messagesystems.com/browse/FE-781)
- [FE-782](https://jira.int.messagesystems.com/browse/FE-782)

### What Changed
- Updates Travis CI config to run Cypress integration tests in parallel with Jest testing
- Updates readme to include new test scripts (along with Prettier reformatting)
- Renamed `blackListHosts` -> `blacklistHosts` to *actually* block requests to outside services like Pendo and Google Analytics
- Adds wide default viewport to Cypress tests to ensure elements are consistently in view
- Adds `__ssid` and `auth` cookies to Cypress whitelist so they are not cleared between tests (this allows the `cy.login()` to be skipped between tests)
- Splits up Cypress tests in to two categories/subfolders - `e2e` and `integration`
- Removes `start-server-and-test` package in favor of `wait-on`
- Removes `http-server` in favor of the development server - `http-server` was not handing routing correctly, making direct `cy.visit()` requests impossible
- Upgrades Cypress to latest version
- Adds screen reader only `<label>` to the `FilterBox` component to improve accessibility and testability of table collection instances
- Adds integration tests for logging in, logging out, signing up, and the templates list page (this was initially used to help get the integration tests working!)
- Updates values for various page titles to more accurately line up with expectations (the page `<title/>` should almost always correspond 1:1 with the top level heading on the page)
- Updates obsolete snapshots introduced due to Prettier formatting
- Adds build verification job to Travis CI

### How To Test
- Verify tests pass via the following methods:
  + Run `npm run test-integration`
  + Run `npm run test-integration-headless`
  + Verify that tests pass in Travis CI
- Verify that documentation makes sense and doesn't have typos

### To Do
